[Shao Yi] Duke Increments

shaoyi1997 opened the pull request.

j-lum committed:
Add support for Gradle workflow

j-lum committed:
Add sample checkstyle configuration

Add toolVersion block in to Gradle code sample to prevent errors.

j-lum committed:
Change file mode on `gradle` to be executable

j-lum committed:
Merge pull request #12 from j-lum/gradle+x

Change file mode on `gradle` to be executable (#9)

shaoyi1997 committed:
Level-1

shaoyi1997 committed:
Level-2

shaoyi1997 committed:
Level-3

shaoyi1997 committed:
Level-4

shaoyi1997 committed:
Level-5

shaoyi1997 committed:
Level-6

shaoyi1997 committed:
Level-7

shaoyi1997 committed:
Level-8

shaoyi1997 committed:
resolved and merge master with branch-level-8

shaoyi1997 committed:
added load from file feature missing from Level 7 criteria

shaoyi1997 committed:
A-MoreOOP

shaoyi1997 committed:
A-JUnit

shaoyi1997 committed:
A-JavaDoc

shaoyi1997 committed:
Level-9

shaoyi1997 committed:
Merge tag 'Level-9'

dalsontws commented on src/main/java/AddCommand.java:
```diff
@@ -0,0 +1,88 @@
+            if (input.length() >= 4) {
+
+                // add Deadline
+                if (input.length() >= 8 && input.substring(0, 8).equals("deadline")) {
```
These conditions can work, but maybe you can try to use a .contains function?
for e.g.
input.contains("deadline")
This can simplify the entire conditional statement.

dalsontws commented on src/main/java/AddCommand.java:
```diff
@@ -0,0 +1,88 @@
+                }
+
+                // add Event
+                else if (input.length() >= 5 && input.substring(0, 5).equals("event")) {
```
Same as line 46.

dalsontws commented on src/main/java/AddCommand.java:
```diff
@@ -0,0 +1,88 @@
+                }
+
+                // add ToDos
+                else if (input.substring(0, 4).equals("todo")) {
```
Same as line 46.

dalsontws commented on src/main/java/AddCommand.java:
```diff
@@ -0,0 +1,88 @@
+     * @param ipt the string representation of the command input.
+     */
+    public AddCommand(String ipt) {
+        input = ipt;
```
Not compulsory, but would be good to have a 'this' keyword to reference the current object. 
e.g. this.input = ipt
Would become ambiguous if there are similar variable names.

dalsontws commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,40 @@
+     */
+    public Deadline(String des, String by) {
+        super(des);
+        this.by = by;
```
"this" keyword used for your tasks constructors, but some methods not used. 
Good to be consistent with the usage!

dalsontws commented on src/main/java/DeleteCommand.java:
```diff
@@ -0,0 +1,49 @@
+     * @param taskNum the index of the task to be deleted
+     */
+    public DeleteCommand(int taskNum) {
+        taskNumber = taskNum;
```
Same as the addCommand method in the AddCommand Class

dalsontws commented on src/main/java/DoneCommand.java:
```diff
@@ -0,0 +1,48 @@
+     * @param taskNum the index of the task to be marked done.
+     */
+    public DoneCommand(int taskNum) {
+        taskNumber = taskNum;
```
Same as the addCommand method in the AddCommand Class

dalsontws commented on src/main/java/FindCommand.java:
```diff
@@ -0,0 +1,57 @@
+     * @param ipt the string of search query.
+     */
+    public FindCommand(String ipt) {
+        input = ipt;
```
Same as the addCommand method in the AddCommand Class

dalsontws commented on src/main/java/AddCommand.java:
```diff
@@ -0,0 +1,88 @@
+     *
+     * @return false.
+     */
+    public boolean isExit() { return false; }
```
Would be clearer and improve readability to have your return statement on a new line!
Maybe you can try running the CheckStyle tool? 
Would help to identify these issues!

shaoyi1997 committed:
Refactor Parser and enforced coding standards

shaoyi1997 committed:
Merge branch 'gradle'

shaoyi1997 committed:
A-Gradle

shaoyi1997 committed:
changed gradle run task to support System.in

shaoyi1997 committed:
A-CheckStyle

shaoyi1997 committed:
Merge branch 'branch-A-CheckStyle'

shaoyi1997 committed:
Add JavaFX functionality Part 1 and 2

shaoyi1997 committed:
Added JavaFX functionality Part 3

shaoyi1997 committed:
fixed bug where nonsense input with length less than 4 results in error

shaoyi1997 committed:
Merge branch 'branch-A-Gradle'

shaoyi1997 committed:
integrated JavaFX, fixed bugs related to find command and loading list from data file

shaoyi1997 committed:
Merge branch 'branch-A-JavaFX'

wilfredbtan commented on src/main/java/DeleteCommand.java:
```diff
@@ -0,0 +1,53 @@
+    public String execute(TaskList tasks, Ui ui, Storage storage) throws DukeException {
+        try {
+            if (taskNumber > tasks.getNumOfTask() || taskNumber < 1) {
+                throw new DukeException("List only has " + tasks.getNumOfTask() + " items.");
```
Should the DukeException be thrown within this try catch block within an if statement? Logic might be confusing. Maybe you can consider a way to catch this error and throw it below as well.

wilfredbtan commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,71 @@
+            return new ExitCommand();
+        } else if (input.length() <= 3) {
+            throw new DukeException("I'm sorry, but I don't know what that means :-(");
+        } else if (input.substring(0, 4).equals("find")) {
```
Is there a better method to check whether a string "contains" something? Also does this catch the errors resulting from invalid inputs for find, done, delete, etc?

shaoyi1997 committed:
Integrated FXML

shaoyi1997 committed:
A-Assertions

shaoyi1997 committed:
Merge pull request #3 from shaoyi1997/branch-A-Assertions

Merge with A-Assertions

shaoyi1997 committed:
A-CodingQuality, refactored and packaged classes

shaoyi1997 committed:
Merge branch 'branch-A-CodingQuality'

shaoyi1997 committed:
merge with branch-A-CodingQuality

shaoyi1997 committed:
fixed conflict in merge with branch-A-CodingQuality

shaoyi1997 committed:
A-Travis

shaoyi1997 committed:
C-Update

shaoyi1997 committed:
Added clone command

shaoyi1997 committed:
Refactored UpdateCommand class, fixed text overrun in GUI

shaoyi1997 committed:
A-UserGuide, changed find command to be case-insensitive, fixed update command when no field is given

shaoyi1997 committed:
Set theme jekyll-theme-architect

shaoyi1997 committed:
Rename UserGuide.adoc to README.adoc

shaoyi1997 committed:
Converted userguide to index.html, disabled gradle test in travis

shaoyi1997 committed:
Merge branch 'master' of https://github.com/shaoyi1997/duke

shaoyi1997 committed:
A-Release, adhered to checkstyle, changed GUI appearance, code cleanup

shaoyi1997 committed:
Delete Ui.PNG

uploaded new Ui.png

Update README.adoc

change build.gradle to accept T&C of gradle

shaoyi1997 committed:
fixed gradle build file

shaoyi1997 committed:
fixed travis status 127 error

shaoyi1997 committed:
Update README.md

shaoyi1997 committed:
updated UserGuide

shaoyi1997 committed:
Merge branch 'master' of https://github.com/shaoyi1997/duke
